internal/mesh/internal/mappers/workloadselectionmapper/workload_selection_mapper.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package workloadselectionmapper
```

Comment:
I generalized this mapper as I needed the same functionality for both explicit destinations and proxy config controllers.